src/pyuvdata/utils/phase_center_catalog.py

```diff
@@ -196,7 +215,13 @@ def look_in_catalog(
                             tol_dict[key][1],
                         )
             else:
-                cat_diffs += check_dict[key] is not None
+                check_type = check_dict.get("cat_type")
+                check_value = None
+                # if type is known, avoid unrelated fields
+                if (check_type is None) or (key not in typenonfields_dict[check_type]):
+                    check_value = check_dict[key]
+                cat_diffs += check_value is not None
+
```

Comment:
I think you can avoid basically all of the additional code by replacing the _old_ L199 with the following:

```suggestion
                cat_diffs += check_dict.get(key, None) is not None
```